File: app/src/main/java/com/example/assignment/RentalRecord.java
package com.example.assignment;

public class RentalRecord {
    private String username;
    private String password; // Add password field
    private String itemName;
    private String type; // "Lend" or "Borrow"
    private String status; // "Available", "Completed", etc.
    private String description;
    private String distance;
    private int credit;
    private int like;
    private int favor;
    private String gmail;
    private String gender; // Added gender field

    public RentalRecord() {}

    public RentalRecord(String username, String itemName, String type, String status, String description, String distance, int credit) {
        this.username = username;
        this.itemName = itemName;
        this.type = type;
        this.status = status;
        this.description = description;
        this.distance = distance;
        this.credit = credit;
        this.like = 0; // Default
        this.favor = 0; // Default
        this.gender = ""; // Default
    }

    // Getters and Setters
    public String getUsername() { return username; }
    public void setUsername(String username) { this.username = username; }

    public String getPassword() { return password; } // Add password getter
    public void setPassword(String password) { this.password = password; } // Add password setter

    public String getItemName() { return itemName; }
    public void setItemName(String itemName) { this.itemName = itemName; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getDistance() { return distance; }
    public void setDistance(String distance) { this.distance = distance; }

    public int getCredit() { return credit; }
    public void setCredit(int credit) { this.credit = credit; }

    public int getLike() { return like; }
    public void setLike(int like) { this.like = like; }

    public int getFavor() { return favor; }
    public void setFavor(int favor) { this.favor = favor; }

    public String getGmail() { return gmail; }
    public void setGmail(String gmail) { this.gmail = gmail; }

    public String getGender() { return gender; } // Added getter
    public void setGender(String gender) { this.gender = gender; } // Added setter

    public String getDisplayText() {
        return type + " \"" + itemName + "\" (" + status + ") - " + distance;
    }
}

File: app/src/main/java/com/example/mad_gruop_ass/RentalRecord.kt
package com.example.mad_gruop_ass

data class RentalRecord(
    var username: String = "",
    var password: String = "", // Add password field
    var itemName: String = "",
    var type: String = "", // "Lend" or "Borrow"
    var status: String = "", // "Available", "Completed", etc.
    var description: String = "",
    var distance: String = "",
    var credit: Int = 0,
    var like: Int = 0,
    var favor: Int = 0,
    var gmail: String = "",
    var gender: String = "" // Added gender field
) {
    constructor(
        username: String,
        itemName: String,
        type: String,
        status: String,
        description: String,
        distance: String,
        credit: Int
    ) : this(
        username = username,
        itemName = itemName,
        type = type,
        status = status,
        description = description,
        distance = distance,
        credit = credit,
        like = 0, // Default
        favor = 0, // Default
        gender = "" // Default
    )

    fun getDisplayText(): String {
        return "$type \"$itemName\" ($status) - $distance"
    }
}


